### PR TITLE
feature: add model and predictor for HuggingFace

### PR DESCRIPTION
**Pull request by icywang86rui**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/huggingface/model.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    This is able to serialize Python lists, dictionaries, and numpy arrays to
+    multidimensional tensors for HuggingFace inference.
+    """
+
+    def __init__(
+        self,
+        endpoint_name,
+        sagemaker_session=None,
+        serializer=NumpySerializer(),
+        deserializer=NumpyDeserializer(),
```
Comment (philschmid) [inline src/sagemaker/huggingface/model.py:44]: `NumpySerializer` -> `JSONSerializer`
```
from sagemaker.serializers import JSONSerializer
from sagemaker.deserializers import JSONDeserializer
```

File: src/sagemaker/huggingface/model.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+
+class HuggingFaceModel(FrameworkModel):
+    """An PyTorch SageMaker ``Model`` that can be deployed to a SageMaker ``Endpoint``."""
+
+    _framework_name = "huggingface"
+
+    def __init__(
+        self,
+        model_data,
```
Comment (philschmid) [inline src/sagemaker/huggingface/model.py:93]: could this be optional, since we have the option to create an endpoint with `model_data`.

**Comment by philschmid** [comment]
Can we also add the `HuggingFaceModel` to the `__init__.py` to make imports easier?

File: src/sagemaker/image_uri_config/huggingface.json [hunk 3]
```diff
@@ -139,5 +139,80 @@
                 }
             }
         }
+    },
+
+    "inference": {
+        "processors": ["gpu"],
```
Comment (saimidu) [inline src/sagemaker/image_uri_config/huggingface.json:145]: We also have `cpu` inference images.
```suggestion
        "processors": ["cpu", "gpu"],
```

File: src/sagemaker/image_uri_config/huggingface.json [hunk 4]
```diff
@@ -139,5 +139,80 @@
...
+        "version_aliases": {
+            "4.6": "4.6.1"
+        },
+        "versions": {
+            "4.6.1": {
+                "version_aliases": {
+                    "pytorch1.6": "pytorch1.7.1",
```
Comment (saimidu) [inline src/sagemaker/image_uri_config/huggingface.json:152]:
```suggestion
                    "pytorch1.7": "pytorch1.7.1",
```

**Pull request closed without merging**